uefi/firmwarevolume.go

```diff
@@ -189,6 +190,39 @@ func (fv *FirmwareVolume) Assemble() ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
+	// Make sure we don't go over the size.
+	fvLen := uint64(len(fv.buf))
+	fOffset := fv.DataOffset
+	for _, f := range fv.Files {
+		fBuf, err := f.Assemble()
+		if err != nil {
+			return nil, err
+		}
+		fLen := uint64(len(fBuf))
+		if fLen+fOffset > fvLen {
```

Comment:
Should this check be moved until after the file is aligned because it might be possible that aligning the file might overflow the FV?

Reply:
Nice catch! I'll change it